src/app/clusters/operational-credentials-server/operational-credentials-cluster.cpp

```diff
@@ -14,62 +14,56 @@
-#include <app-common/zap-generated/ids/Attributes.h>
-#include <app/AttributeAccessInterface.h>
-#include <app/AttributeAccessInterfaceRegistry.h>
-#include <app/CommandHandler.h>
-#include <app/ConcreteCommandPath.h>
 #include <app/EventLogging.h>
 #include <app/InteractionModelEngine.h>
+#include <app/data-model-provider/MetadataTypes.h>
 #include <app/reporting/reporting.h>
-#include <app/server/Dnssd.h>
+#include <app/server-cluster/AttributeListBuilder.h>
 #include <app/server/Server.h>
-#include <app/util/attribute-storage.h>
-#include <clusters/OperationalCredentials/Attributes.h>
+#include <clusters/OperationalCredentials/AttributeIds.h>
 #include <clusters/OperationalCredentials/Commands.h>
-#include <clusters/OperationalCredentials/Events.h>
+#include <clusters/OperationalCredentials/Enums.h>
 #include <clusters/OperationalCredentials/Metadata.h>
-#include <clusters/OperationalCredentials/Structs.h>
 #include <credentials/CHIPCert.h>
 #include <credentials/CertificationDeclaration.h>
 #include <credentials/DeviceAttestationConstructor.h>
-#include <credentials/DeviceAttestationCredsProvider.h>
-#include <credentials/FabricTable.h>
-#include <credentials/GroupDataProvider.h>
-#include <lib/core/CHIPSafeCasts.h>
-#include <lib/core/PeerId.h>
 #include <lib/support/CodeUtils.h>
-#include <lib/support/ScopedBuffer.h>
-#include <lib/support/logging/CHIPLogging.h>
-#include <platform/CHIPDeviceLayer.h>
-#include <string.h>
 #include <tracing/macros.h>
 
 using namespace chip;
-using namespace ::chip::Transport;
 using namespace chip::app;
 using namespace chip::app::Clusters;
 using namespace chip::app::Clusters::OperationalCredentials;
 using namespace chip::Credentials;
-using namespace chip::Crypto;
 using namespace chip::Protocols::InteractionModel;
-
 namespace {
 
 constexpr auto kDACCertificate = CertificateChainTypeEnum::kDACCertificate;
 constexpr auto kPAICertificate = CertificateChainTypeEnum::kPAICertificate;
 
-void SendNOCResponse(app::CommandHandler * commandObj, const ConcreteCommandPath & path, NodeOperationalCertStatusEnum status,
-                     uint8_t index, const CharSpan & debug_text);
-NodeOperationalCertStatusEnum ConvertToNOCResponseStatus(CHIP_ERROR err);
+// Get the attestation challenge for the current session in progress. Only valid when called
+// synchronously from inside a CommandHandler. If not called in CASE/PASE session context,
+// return an empty span. This will for sure make the procedures that rely on the challenge
+// fail, which is intended as it never should have reached here.
+// TODO: Create an alternative way to retrieve the Attestation Challenge without this huge amount of calls.
+ByteSpan GetAttestationChallengeFromCurrentSession(app::CommandHandler * commandObj)
+{
+    VerifyOrDie((commandObj != nullptr) && (commandObj->GetExchangeContext() != nullptr));
+    Transport::Session::SessionType sessionType = commandObj->GetExchangeContext()->GetSessionHandle()->GetSessionType();
```

Comment:
should we store `commandObj->GetExchangeContext()->GetSessionHandle()` as a variable since we call it twice?

Reply:
Stored the SessionHandle in variable instead of calling it twice.